backend/overstroomik_service/geoserver.py

```diff
@@ -0,0 +1,152 @@
+
+                    out = result.json()
+                    features = out.get("features")
+
+                    if len(features) > 0:
+
+                        for feature in features:
+
+                            # id or name of the sub layer
+                            f_id = feature.get("id")
+                            properties = feature.get("properties")
+
+                            if len(f_id) == 0 or f_id.startswith(
+                                    settings.LAYER_MAXIMUM_WATER_DEPTH):
+                                data_item["maximum_water_depth"] = properties.get(
+                                    settings.FIELD_MAXIMUM_WATER_DEPTH, None)
+
+                            elif f_id.startswith(settings.LAYER_SAFETY_BOARD_ID):
+                                data_item["safety_board_id"] = properties.get(
+                                    settings.FIELD_SAFETY_BOARD_ID, None)
+
+                            elif f_id.startswith(
+                                    settings.LAYER_PROBABILITY_OF_FLOODING):
+                                data_item["probability_of_flooding"] = properties.get(
+                                    settings.FIELD_PROBABILITY_OF_FLOODING, None)
+
+                            elif f_id.startswith(settings.LAYER_EVACUATION_PERCENTAGE):
+                                data_item["evacuation_percentage"] = properties.get(
+                                    settings.FIELD_EVACUATION_PERCENTAGE, None)
+
+                            elif f_id.startswith(settings.LAYER_FLOOD_TYPE):
+                                data_item["flood_type"] = properties.get(
+                                    settings.FIELD_FLOOD_TYPE, None)
+
+                        else:
+                            status: ERROR_GEOS_NO_SMAP
+                else:
+                    status = ERROR_GEOS_NO_RESP
+
+        return status, Data(**data_item)
+
+    def get_api_url_from_rd(self,
+                            rd_x: float,
+                            rd_y: float,
+                            geoserver_url: Optional[str] = settings.GEOSERVER_URL,
+                            layers: [str] = settings.GEOSERVER_LAYER):
+        """
+        Create the get-feature-info link.
+        :param rd_x: x coordinate in meters, geocoded location longitude transformed into EPSG:28992
+        :param rd_y: y coordinate in meters, geocoded location latitude transformed into EPSG:28992
+        :param geoserver_url: link to the geoserver (example: http://geoserver:8080/geoserver)
+        :param layers: group layer with the expected data  (example: overstroomik:Overstroomik_data)
+        """
+
+        # api url template
+        api_get_feature_info = f"{geoserver_url}/overstroomik/wms?SERVICE=WMS&VERSION=1.1.1"\
+            f"&REQUEST=GetFeatureInfo&INFO_FORMAT=application/json&SRS=EPSG:28992&FEATURE_COUNT=50"\
+            f"&LAYERS={layers}&QUERY_LAYERS={layers}"
+
+        # bounding box
```

Comment:
Document that a getfeatureinfo requires a bbox+width+height, so we fully control/can calculate the correct indices ourselves, which is why this can be hardcoded.

Reply:
Additional comments added